### PR TITLE
sdk: allow passing docker build arguments in service manifest

### PR DESCRIPTION
**Pull request by remcoros**
note: I snuck in the silencing of mksquashfs/tar2squashfs output. I can remove that from this PR and move it to another.

manifest.ts

```
{
  images: {
    main: {
      source: {
        dockerBuild: {
          workdir: '.',
          dockerfile: 'Dockerfile',
          buildArgs: {
            TEST: "test",
            TEST_FROMENV: { env: "FROMENV" },
          }
        },
      },
      arch: ['x86_64'],
    },
  },
}
```

Dockerfile

```
FROM baseimage

# set automatically by 'docker build'
ARG TARGETPLATFORM
ARG TARGETARCH

# set by 'start-cli s9pk pack', matching currently build 'arch'
ARG ARCH

# static value from manifest.ts
ARG TEST
# from ENV var, specified in manifest.ts
ARG TEST_FROMENV
```

Makefile:

```
${PACKAGE_ID}.s9pk: $(shell start-cli s9pk list-ingredients) docker_entrypoint.sh
	FROMENV=FromMakefile \
	start-cli s9pk pack
```

### CODE REVIEW

**Review by Blu-J — APPROVED** [review]
Looks simple mostly. Only thing not trivial is the build-arg added, but looks like it should work. Would love to have a screenshot or something to show it works.

**Comment by remcoros** [comment]
image

the added ARCH arch is more of a convenience, it is used in pretty much all existing wrappers, and since the currently building 'arch' is already available, it was easy to pass through.

File: core/startos/src/s9pk/v2/pack.rs [hunk 1]
```diff
@@ -337,6 +341,22 @@ impl clap::FromArgMatches for ImageConfig {
...
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, TS)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+#[ts(export)]
+pub enum BuildArg {
+    String(String),
+    EnvVar(EnvVar),
+}
```
Comment (dr-bonez) [inline core/startos/src/s9pk/v2/pack.rs:358]:
```suggestion
#[derive(Debug, Clone, Deserialize, Serialize, TS)]
#[serde(rename_all = "camelCase")]
#[serde(untagged)]
#[ts(export)]
pub enum BuildArg {
    String(String),
    EnvVar { 
        env: String,
    }
}
```
I don't think EnvVar warrants its own struct type, given that it is highly contextual

Comment (remcoros) [reply core/startos/src/s9pk/v2/pack.rs]: much better 👍

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**